### PR TITLE
[react-window] Update typings to 1.5.1

### PR DESCRIPTION
**Pull request by heyimalex**
This is an update to the [react-window](https://github.com/bvaughn/react-window) typings, mostly minor things but there have been some [changes](https://github.com/bvaughn/react-window/blob/master/CHANGELOG.md) and they're all reflected here. CC @martynaskadisa

### CODE REVIEW

**Review by martynaskadisa — APPROVED** [review]
Thanks for your time improving these typings. LGTM 👍

File: types/react-window/index.d.ts [hunk 1]
```diff
@@ -1,10 +1,19 @@
-// Type definitions for react-window 1.1
+// Type definitions for react-window 1.5
 // Project: https://github.com/bvaughn/react-window/
 // Definitions by: Martynas Kadiša <https://github.com/martynaskadisa>
+//                 Alex Guerra <https://github.com/heyimalex>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 3.0
```
Comment (minestarks) [inline types/react-window/index.d.ts]: Is there a reason for the TS 3.0 dependency?

Comment (heyimalex) [reply types/react-window/index.d.ts]: I just blindly followed the recommendation from the [logs from the ci failure](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/490209189#L2520).

> Compile error in typescript@2.9 but not in typescript@3.0.
> Fix with a comment '// TypeScript Version: 3.0' just under the header.
> Class 'Component<ListChildComponentProps, {}, any>' defines instance member function 'shouldComponentUpdate', but extended class 'RowWithShouldComponentUpdate' defines it as instance member property.

It's triggered by this code. react-window exports a custom `shouldComponentUpdate` and says it should be it be used like this.

```typescript
import { shouldComponentUpdate } from "react-window";
class RowWithShouldComponentUpdate extends React.Component<
    ListChildComponentProps
> {
    shouldComponentUpdate = shouldComponentUpdate.bind(this);
    render() {
        const { index, style } = this.props;
        return <div style={style}>Row {index}</div>;
    }
}
```

This effects _using_  `shouldComponentUpdate`, but the definition might be fine with 2.8. But I'm not sure how to get it to pass CI without removing the test or depending on 3.0.

Comment (minestarks) [reply types/react-window/index.d.ts]: I see, subtle!
TypeScript 2.9 sounds too nitpicky. You could change the test code to something like this to satisfy the CI without forcing a TS version upgrade. 

```ts
shouldComponentUpdate(...args: any[])  {
        return shouldComponentUpdate.call(this, ...args);
    }
```

Comment (heyimalex) [reply types/react-window/index.d.ts]: Thanks! I updated the test and changed `TypeScript Version` back to 2.8.

**Pull request merged**